core/chain_makers_test.go

```diff
@@ -97,4 +101,302 @@ func ExampleGenerateChain() {
 	// balance of addr1: 989000
 	// balance of addr2: 10000
 	// balance of addr3: 19687500000000001000
+} */
+
+// Runner function
+func chainsValidator(chain []*types.Block, primeChain BlockChain, regionChain BlockChain, zoneChain BlockChain, specsPool []blockSpecs) (validNetwork [3]BlockChain) {
+	// pass blocks in through all possible methods to verify similar handling
+	// e.g. rpc
+
+	// handle permutation *outside* this function
+	// will need to hit this function twice - first to create blocks (with tags), second to test tag scenarios
+
+	if _, err := primeChain.InsertChain(chain); err != nil {
+		print(err)
+	}
+
+	validNetwork = [3]BlockChain{primeChain, regionChain, zoneChain}
+
+	return validNetwork
+}
+
+// struct for notation
+type blockConstructor struct {
+	numbers [3]int // prime idx, region idx, zone idx
+	// -1 for contexts the block does not coincide with
+	tag        int    // specify what tag to construct block in (0 must be first chain grinded)
+	parentTags [3]int // index specifies whether to look for Prime, Region, or Zone order block in that respective tag
+	// parentTags    [3]string // (optionally) Override the parents to point to tagged blocks. Empty strings are ignored.
+	// tag           string    // (optionally) Give this block a named tag. Empty strings are ignored.
```

Comment:
Please leave tags as strings. Integers do not suffice. For example, I may have a fork in a region, resulting in two region blocks claiming to be block # 5. If I want to create a twist, which references the region back block 5 in the region, how can you know which block I am referencing back to? The only way to do this, is to provide a unique reference/tag to the block you wish to refer back to.